### PR TITLE
Admin monitor message: separation between content and display

### PR DESCRIPTION
**Pull request by Wadeck**
- separate the anchors from the message in my recent monitors
- add dismiss button for them
- use `p` instead of `br`
- put the version for the `RootUrlNotSetMonitor` and the `HttpResponses` new method

No issue related, @recena that points me to that problem.

### Screenshots

For the root URL monitor message:
root_url_warning

For the CSRF monitor message:
csrf_warning

### Proposed changelog entries

* Allow the "CSRF protection missing" and "Root URL not set" administrative monitors to be dismissed directly from their warning messages.

### Desired reviewers

@reviewbybees @recena

### CODE REVIEW

File: core/src/main/java/jenkins/diagnostics/RootUrlNotSetMonitor.java [hunk 1]
```diff
@@ -52,15 +52,13 @@ public String getDisplayName() {
...
-        );
+        return loc.getUrl() == null || !UrlHelper.isValidRootUrl(loc.getUrl());
     }
     
     // used by jelly to determined if it's a null url or invalid one
     @Restricted(NoExternalUse.class)
     public boolean isUrlNull(){
         JenkinsLocationConfiguration loc = JenkinsLocationConfiguration.get();
-        return loc != null && loc.getUrl() == null;
+        return loc.getUrl() == null;
```
Comment (dwnusbaum) [inline core/src/main/java/jenkins/diagnostics/RootUrlNotSetMonitor.java]: If you don't need the null check on `loc` here than can you remove it from `isActivated` as well? Looks like it is `@CheckForNull` so I think you want to add it back.

File: core/src/main/resources/jenkins/diagnostics/RootUrlNotSetMonitor/message.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
 urlIsNull=Jenkins root URL is empty but is required for the proper operation of many Jenkins features like email notifications, \
-PR status update, and environment variables such as <code>BUILD_URL</code>.<br />\
-Please provide an accurate value in <a href="{0}/configure">Jenkins configuration</a>.
+PR status update, and environment variables such as <code>BUILD_URL</code>.
 urlIsInvalid=Jenkins root URL seems to be invalid. It is required for the proper operation of many Jenkins features like email notifications, \
-PR status update, and environment variables such as <code>BUILD_URL</code>.<br />\
-Please provide an accurate value in <a href="{0}/configure">Jenkins configuration</a>.
+PR status update, and environment variables such as <code>BUILD_URL</code>.
+actionToTake=Please provide an accurate value in {0}.
+actionUrlContent=Jenkins configuration
```
Comment (dwnusbaum) [inline core/src/main/resources/jenkins/diagnostics/RootUrlNotSetMonitor/message.properties]: nit: add a newline

File: core/src/main/java/jenkins/diagnostics/RootUrlNotSetMonitor.java (hunk 1 above)
Comment (Wadeck) [reply core/src/main/java/jenkins/diagnostics/RootUrlNotSetMonitor.java]: It's already the case for `isActivated` and the `JenkinsLocationConfiguration.get()` is `@Nonnull`.

Do you have the lastest version or perhaps it's me

Comment (dwnusbaum) [reply core/src/main/java/jenkins/diagnostics/RootUrlNotSetMonitor.java]: I was looking at an old version, this looks good to me. Sorry for the noise!

File: core/src/main/resources/jenkins/security/csrf/CSRFAdministrativeMonitor/message.jelly [hunk 3]
```diff
@@ -22,8 +22,19 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 -->
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core">
+<j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form">
     <div class="alert alert-warning">
-        ${%warningMessage(rootURL)}
+        <form method="post" action="${rootURL}/${it.url}/disable">
```
Comment (dwnusbaum) [inline core/src/main/resources/jenkins/security/csrf/CSRFAdministrativeMonitor/message.jelly]: Is allowing this monitor to be dismissed important enough for a changelog entry?

**Review by dwnusbaum — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

**Comment by recena** [comment]
Much better.

**Review by recena — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
👍

**Comment by dwnusbaum** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
@Wadeck please write a changelog entry for it

**Comment by Wadeck** [comment]
@oleg-nenashev here you are! 👍

**Comment by dwnusbaum** [comment]
@Wadeck I edited it slightly for clarity!

**Pull request merged**